### PR TITLE
Scala 3

### PR DESCRIPTION
**Pull request by mtomko**
https://github.com/broadinstitute/poolq/issues/5

### CODE REVIEW

File: src/main/scala/org/broadinstitute/gpp/poolq3/PoolQConfig.scala [hunk 1]
```diff
@@ -388,7 +388,7 @@ object PoolQConfig {
     // run control
     args += (("row-matcher", config.rowMatchFn))
     args += (("col-matcher", config.colMatchFn))
-    if (config.countAmbiguous) {
+    if config.countAmbiguous then {
```
Comment (nessus42) [inline src/main/scala/org/broadinstitute/gpp/poolq3/PoolQConfig.scala:391]: Braces with `then`?

**Comment by nessus42** [comment]
Personally, if I were going to switch to `then` in if-then-else's, I wouldn't also use curly braces. I'd go completely new-style or stay completely old-style wrt this. I find the mixture to be a bit discombobulating.

OTOH, I have no interest in being a style nazi either.

:cat: :moon: 🎱

**Comment by mtomko** [comment]
I just run the compiler through with the first set of options described here:

https://docs.scala-lang.org/scala3/guides/migration/tooling-syntax-rewriting.html#the-new-syntax-rewrites

Namely, `-new-syntax -rewrite`. I manually replaced imports ending with `._` with `.*`, too.

Because Scala 3's syntactic changes are so new, I decided not to go to the indentation significant route yet. `if-then-else` plus `for-do` etc has the advantage (in my opinion) of having fewer parens, which offers some moderate value.

I may well go the route of significant indentation here at some point, though. If people feel like PoolQ should go whole hog, I'm happy to run the next set of compiler syntax rewrites. Perhaps I'll do that anyway and provide it as a commit at the end here, which we could decide to keep or drop.

**Review by mtomko — COMMENTED** [review]
I rolled back the indentation-significant syntax changes. There is a bug in scalafmt that I can't work around. See my comment on  

https://github.com/scalameta/scalafmt/issues/3653

That issue is closed, I'm going to either open a new one or get them to re-open that one.

**Review by tmgreen — APPROVED** [review]
👍

**Comment by kitbellew** [comment]
> I just run the compiler through with the first set of options described here:
> 
> https://docs.scala-lang.org/scala3/guides/migration/tooling-syntax-rewriting.html#the-new-syntax-rewrites

fyi: https://scalameta.org/scalafmt/docs/configuration.html#rewritescala3converttonewsyntax

**Comment by mtomko** [comment]
Closing in favor of https://github.com/broadinstitute/poolq/pull/34

**Pull request closed without merging**